newIDE/app/src/MainFrame/index.js

```diff
@@ -935,6 +936,9 @@ export default class MainFrame extends React.Component<Props, State> {
   save = () => {
     if (!this.state.currentProject) return;
 
+    if (this.state.editorTabs) {
```

Comment:
You should not need this if I think

Reply:
it was failing flow checks without it. Should I re-add it? :)

Reply:
Try to re-add it and let's see the error. :)
If I look at the typing of the state, it's:

```
type State = {|
...
editorTabs: EditorTabsState,
...
|}
```

which means that editorTabs is always properly defined. Which make sense as there is no reason for this component to exist without a list of tabs (even if the list is empty, it's still defined).

Reply:
I've added a comment with the solution (if I'm not mistaken). Even if they are intimidating, always fully read the flow complains, because here it's not related to editorTabs.